tests/connection/broker.cpp

```diff
@@ -33,12 +33,10 @@ class Subscriber
     void run()
     {
         zeroeq::Subscriber subscriber( test::buildURI( "127.0.0.1", *_publisher ));
-        ::test::SerializablePtr echoEvent = ::test::getFBEchoInEvent(
-                    [this]( const zeroeq::FBEvent& event )
-                    { test::onEchoEvent( event );
-                      received = true; });
 
-        BOOST_CHECK( subscriber.subscribe( *echoEvent ));
+        BOOST_CHECK( subscriber.subscribe( test::Echo::IDENTIFIER(),
+             zeroeq::EventPayloadFunc([&]( const void* data, const size_t size )
```

Comment:
You can pass the lambda directly as the argument without invoking the zeroeq::EventPayloadFunc constructor.